parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -362,6 +367,16 @@ public ReadBuilder createReaderFunc(Function<MessageType, ParquetValueReader<?>>
       return this;
     }
 
+    public ReadBuilder createBatchedReaderFunc(Function<MessageType, VectorizedReader<?>> func) {
+      this.batchedReaderFunc = func;
+      return this;
+    }
+
+    public ReadBuilder enableBatchedRead() {
```

Comment:
Why not enable batch read when `createBatchReaderFunc` is called? If that's null, then use the other read path. If it is defined, then use the vectorized path.

Reply:
Done